butterfree/transform/feature_set.py

```diff
@@ -261,13 +264,12 @@ def get_schema(self) -> List[Dict]:
 
         for f in self.features:
             name = self._get_features_columns(f)
-            if f.transformation:
-                windows = f.transformation._windows or [None]
+            if isinstance(f.transformation, SparkFunctionTransform):
```

Comment:
We can just verify the Transform abstract class? I mean, if we implement a new Transform we would have to update here again?

Reply:
I see... So the logic here is a little bit complicated. Every change in our existing entities, we will have to change this `Metadata` class. Besides that, new entities probably will not work properly with Metadata too, like new Transformers, new Readers etc.

Reply:
I think this fix works for now obviously, but in my opinion, we will need to discuss a way to refactor this feature to make things easier and more scalable. One idea for example:
For this class to be able to be more generalistic each block that builds the pipeline should be responsible for its own metadata. So the `Metadata` entity just calls something like a method `.get_metadata()` or a property `.metadata`. In that way, the `_construct` would just need to iterate through all the blocks that built the pipeline and get the metadata from each piece, so it would be more agnostic to the specialized types and we wouldn't need a lot of verifications.